src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -0,0 +1,89 @@
+package net.sf.jabref.model.database;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fileformat.BibtexParser;
+import net.sf.jabref.model.entry.BibEntry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+
+public class BibDatabaseTest {
+
+    @Before
+    public void setUp() {
+        Globals.prefs = JabRefPreferences.getInstance(); // set preferences for this test
+    }
+
+    @After
+    public void tearDown() {
+        Globals.prefs = null;
+    }
+
+    /**
+     * Some basic test cases for resolving strings.
+     *
+     * @throws FileNotFoundException
+     * @throws IOException
+     */
+    @Test
+    public void testResolveStrings() throws IOException {
+        try (FileReader fr = new FileReader("src/test/resources/net/sf/jabref/util/twente.bib")) {
+            ParserResult result = BibtexParser.parse(fr);
+
+            BibDatabase db = result.getDatabase();
+
+            assertEquals("Arvind", db.resolveForStrings("#Arvind#"));
+            assertEquals("Patterson, David", db.resolveForStrings("#Patterson#"));
+            assertEquals("Arvind and Patterson, David", db.resolveForStrings("#Arvind# and #Patterson#"));
+
+            // Strings that are not found return just the given string.
+            assertEquals("#unknown#", db.resolveForStrings("#unknown#"));
+        }
+    }
+
+
+    /**
+     * Insert two instances of {@link BibEntry} with the same ID into a {@link BibDatabase},
+     * expects a {@link KeyCollisionException}
+     */
+    @Test(expected = KeyCollisionException.class)
+    public void testInsertEntryKeyCollision() {
+        BibDatabase db = new BibDatabase();
+        assertEquals(Collections.emptyList(), db.getEntries());
+
+        BibEntry entry0 = new BibEntry();
+        db.insertEntry(entry0);
+        assertEquals(db.getEntries().size(), 1);
```

Comment:
I second this comment. One test should check that insertEntry really inserts the entry and a second test to check that containsEntryWithId works, finally a third test with 2 items.